gcip/addons/container/jobs/crane.py

```diff
@@ -118,3 +118,65 @@ def push(
         job.prepend_scripts(*docker_client_config.get_shell_command())
 
     return job
+
+
+def pull(
+    src_registry: Union[Registry, str],
+    *,
+    tar_path: Optional[str] = None,
+    image_name: Optional[str] = None,
+    image_tag: Optional[str] = None,
+    docker_client_config: Optional[DockerClientConfig] = None,
+    crane_image: Optional[Union[Image, str]] = None,
+) -> Job:
+    """
+    Creates a job to pull container image from remote container registry with `crane`.
+
+    Args:
+        src_registry (str): Registry URL to pull container image from.
+        image_name (str): Container image with namespace to pull from `src_registry`.
+        image_tag (str): Tag of the image which will be pulled.
+        tar_path (Optional[str], optional): Path where to save the container image tarball.
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_DIR`. Defaults to None.
+        docker_client_config (Optional[DockerClientConfig], optional): Creates the Docker configuration file base on objects settings,
+            to authenticate against given registries. Defaults to a `DockerClientConfig` with login to the official Docker Hub
+            and expecting credentials given as environment variables `REGISTRY_USER` and `REGISTRY_LOGIN`.
+        crane_image (Optional[Union[Image, str]], optional): Container image which contains `crane` command.
+            Defaults to PredefindedImages.CRANE.
+
+    Returns:
+        Job: Returns a `gcip.Job`, with neccessary configuration to pull a container image from a remote registry and stores it in `tar_path`.
+            Job runs in ```stage=pull```
+    """
+    if not crane_image:
+        crane_image = PredefinedImages.CRANE
+
+    if not tar_path:
+        tar_path = PredefinedVariables.CI_PROJECT_DIR
+
+    if not image_name:
+        image_name = PredefinedVariables.CI_PROJECT_NAME
```

Comment:
In which use case is it helpful to default the image name to the current repository? Isn't it more helpful to not default the `image_name` and force the user to provide one? It seems to me a seldom case that the user want to pull an image equally named to the target repo. Thus it might be more helpful to guide the user, that he should provide an image name?
Despite I didn't see the typical use case.

Reply:
You are right. 
The function signature looks like `pull(*, src_registry, image_name, ... Optional parameters)